pilot/pkg/autoregistration/internal/health/controller.go

```diff
@@ -0,0 +1,109 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package health
+
+import (
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"istio.io/api/meta/v1alpha1"
+	"istio.io/istio/pilot/pkg/autoregistration/internal/state"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/model/status"
+	"istio.io/istio/pkg/kube/controllers"
+	istiolog "istio.io/pkg/log"
+)
+
+var log = istiolog.RegisterScope("wle", "wle controller debugging")
+
+type HealthEvent struct {
+	// whether or not the agent thought the target is healthy
+	Healthy bool `json:"healthy,omitempty"`
+	// error message propagated
+	Message string `json:"errMessage,omitempty"`
+}
+
+type HealthCondition struct {
+	proxy     *model.Proxy
+	entryName string
+	condition *v1alpha1.IstioCondition
+}
+
+// Controller knows how to update health status of a workload.
+type Controller struct {
+	stateStore *state.Store
+
+	// healthCondition is a fifo queue used for updating health check status
+	healthCondition controllers.Queue
+}
+
+// NewController returns a new Controller instance.
+func NewController(stateStore *state.Store, maxRetries int) *Controller {
+	c := &Controller{
+		stateStore: stateStore,
+	}
+	c.healthCondition = controllers.NewQueue("healthcheck",
+		controllers.WithMaxAttempts(maxRetries),
+		controllers.WithGenericReconciler(c.updateWorkloadEntryHealth))
+	return c
+}
+
+func (c *Controller) Run(stop <-chan struct{}) {
+	go c.healthCondition.Run(stop)
+	<-stop
```

Comment:
nit:  these are equal to` c.healthCondition.Run(stop)`

Reply:
done